packages/plugin-ext/src/hosted/node/plugin-reader.ts

```diff
@@ -36,15 +36,20 @@ export class HostedPluginReader implements BackendApplicationContribution {
     private readonly scanner: MetadataScanner;
 
     @optional()
-    @multiInject(MetadataProcessor) private readonly metadataProcessors: MetadataProcessor[];
+    @multiInject(MetadataProcessor)
+    private readonly metadataProcessors: MetadataProcessor[];
+
+    @optional()
+    @multiInject(PluginResourcesProvider)
+    private readonly resourcesProvider: PluginResourcesProvider[];
 
     /**
      * Map between a plugin's id and the local storage
      */
     private pluginsIdsFiles: Map<string, string> = new Map();
 
     configure(app: express.Application): void {
-        app.get('/hostedPlugin/:pluginId/:path(*)', (req, res) => {
+        app.get('/hostedPlugin/:pluginId/:path(*)', async (req, res) => {
```

Comment:
I think it would make more sense to configure local resources as a PluginResourceProvider, as well and use a "chain of responsiblity" pattern. Do we ever get local resources for /hostedPlugin/:pluginId/path?

Reply:
>  Do we ever get local resources for /hostedPlugin/:pluginId/path?

always in the vanilla Theia? all VS Code extension's images?

Reply:
> "chain of responsiblity" pattern 

Actually express.js support it already, instead of introducing new apis, we could just call `next` handler for the given request: https://expressjs.com/en/guide/using-middleware.html#middleware.application